### PR TITLE
Fix System.ValueTuple on net 4.7.1   ---- issue 3896

### LINKED ISSUE #3896: System.ValueTuple reference missing from F# compiler default references when using .NET 4.7.1
I'm not sure of the cause but I'm seeing three test failures on my local machine. All three are related to ``System.ValueTuple.dll``.  Note this machine has .NET 4.7.1 installed. 

The individual steps are:

    cd tests\fsharp\typecheck\sigs
    ..\..\release\net40\bin\fsc.exe -r:System.Core.dll --nowarn:20 --define:COMPILED --target:exe -o:pos27.exe pos27.fs

    cd tests\fsharp\core\quotes
    ..\..\release\net40\bin\fsc.exe -r:System.Core.dll --nowarn:20 --define:COMPILED -o:test.exe -r cslib.dll -g test.fsx

Both appear to be corrected by adding ``-r:System.ValueTuple.dll`` to the compilation line.  I think it's just a problem with the test framework

Screenshot of errors below

image

### PR DESCRIPTION
**Pull request by KevinRansom**
This fixes #3896 

What was happening is that on net471 the F# compiler was not fiunding any metadata describing the System.ValueTuple type because.  It was loading the System.ValueTuple from mscorlib.dll.

This change ensures that the compiler will load the type from the System.ValueTuple dll, that is provided either as a command line reference or by loading the System.ValueTuple.dll from all of the places we normally magically look.

It looks like the FCS doesn't require a deployed valuetuple dll, and so we disable default reference for it.
In general that won't be a problem because the standard F# templates reference the System.ValueTuple package anyway and so they are not reliant on this default reference specification.

### CODE REVIEW

**Comment by KevinRansom** [comment]
@dotnet-bot Windows_NT Release_ci_part2 Build please

**Comment by KevinRansom** [comment]
@dotnet-bot test Windows_NT Release_ci_part2 Build please

**Comment by KevinRansom** [comment]
@dotnet-bot test Ubuntu14.04 Release_fcs Build please

**Comment by KevinRansom** [comment]
@dotnet-bot test Windows_NT Release_ci_part2 Build please

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
@dotnet-bot test this

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

File: src/fsharp/CompileOps.fs [hunk 1]
```diff
@@ -1774,20 +1774,25 @@ let OutputDiagnosticContext prefix fileLineFn os err =
 let GetFSharpCoreLibraryName () = "FSharp.Core"
 
 // If necessary assume a reference to the latest .NET Framework FSharp.Core with which those tools are built.
-let GetDefaultFSharpCoreReference() = typeof<list<int>>.Assembly.Location
+let GetDefaultFSharpCoreReference () = typeof<list<int>>.Assembly.Location
 
-// If necessary assume a reference to the latest System.ValueTuple with which those tools are built.
-let GetDefaultSystemValueTupleReference() = 
-#if COMPILER_SERVICE_AS_DLL
```
Comment (KevinRansom) [inline src/fsharp/CompileOps.fs]: @dsyme.
I enabled this reference assembly for FCS.  Let me know if you want this back under an #ifdef.

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
@mmitche do you know why it's not triggering the tests?

Thanks
Kevin

**Comment by mmitche** [comment]
@KevinRansom The servers are moving, there was an outage notification sent a few times over the last week or so.  Did you get it?

**Comment by KevinRansom** [comment]
Oh... I never noticed it.  That makes sense.  When are they back on line?

**Comment by mmitche** [comment]
@KevinRansom I am hoping in a couple hours.

**Comment by KevinRansom** [comment]
Sweet, thanks mate

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
@mmitche is there some reason that our Ubuntu ci tasks aren't triggering?

**Comment by KevinRansom** [comment]
@mmitche  is there some reason that our Ubuntu ci tasks aren't triggering?

File: src/fsharp/CompileOps.fs (hunk 1 above)
Comment (dsyme) [reply src/fsharp/CompileOps.fs]: It's ok I guess.  We should really add FCS documentation on the behaviour you get for default references

**Pull request merged**